sieve_concurrent.go

```diff
@@ -0,0 +1,52 @@
+package eratosthenes
+
+func sequence(i int) <-chan int {
+	seq := make(chan int)
+
+	go func() {
+		// This will obviously leave a locked goroutine
+		for {
+			seq <- i
+			i += 1
+		}
+	}()
+
+	return seq
+}
+
+func filterNotMultiplesOf(n int, seq <-chan int) <-chan int {
+	filtered := make(chan int)
+
+	go func() {
+		for i := range seq {
+			if i%n != 0 {
+				filtered <- i
+			}
+		}
+	}()
+
+	return filtered
+}
+
+func Concurrent(n int) []int {
+	// This algorithm will create O(N) goroutines and channels
```

Comment:
I was thinking about space complexity, but I'm not sure if I could use big O here =P